src/modules/blockchainApplication/explore/components/BlockchainApplications/BlockchainApplications.test.js

```diff
@@ -1,44 +1,55 @@
 import { screen } from '@testing-library/react';
-import mockBlockchainApplications from '@tests/fixtures/blockchainApplicationsExplore';
 import { usePinBlockchainApplication } from '@blockchainApplication/manage/hooks/usePinBlockchainApplication';
-import { renderWithRouter } from 'src/utils/testHelpers';
+import { renderWithRouterAndQueryClient } from 'src/utils/testHelpers';
+import { useBlockchainApplicationExplore } from '../../hooks/queries/useBlockchainApplicationExplore';
+import { useBlockchainApplicationStatistics } from '../../hooks/queries/useBlockchainApplicationStatistics';
+import { mockBlockchainApp, mockBlockchainAppStatistics } from '../../__fixtures__';
 import BlockchainApplications from './BlockchainApplications';
 
-jest.useFakeTimers();
-jest.mock('@blockchainApplication/manage/hooks/usePinBlockchainApplication');
 const mockTogglePin = jest.fn();
-const mockedPins = [mockBlockchainApplications[0].chainID];
+const mockedPins = [mockBlockchainApp.data[0].chainID];
+const mockApplyFilters = jest.fn();
+const mockFetchNextPage = jest.fn();
+
+jest.useFakeTimers();
+jest.mock('../../../manage/hooks/usePinBlockchainApplication');
+jest.mock('../../hooks/queries/useBlockchainApplicationExplore');
+jest.mock('../../hooks/queries/useBlockchainApplicationStatistics');
+jest.mock('@common/hooks', () => ({
+  ...jest.requireActual('@common/hooks'),
+  useFilter: jest.fn(() => ({
+    applyFilters: mockApplyFilters,
+  })),
+}));
```

Comment:
mock a directory has effect on other files, please avoid it and use jest spy on, or even hit the actual hooks

Reply:
@soroushm Resolved